test_rmw_implementation/test/test_serialize_deserialize.cpp

```diff
@@ -57,6 +57,18 @@ TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), serialize_with_b
   EXPECT_NE(RMW_RET_OK, rmw_serialize(&input_message, ts, &serialized_message));
   rmw_reset_error();
 
+  rosidl_message_type_support_t * non_const_ts =
+    const_cast<rosidl_message_type_support_t *>(ts);
+  const char * typesupport_identifier = non_const_ts->typesupport_identifier;
+  non_const_ts->typesupport_identifier = "not-a-typesupport-identifier";
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+  EXPECT_NE(RMW_RET_OK, rmw_serialize(&input_message, non_const_ts, &serialized_message));
+  rmw_reset_error();
+  non_const_ts->typesupport_identifier = typesupport_identifier;
```

Comment:
@Lobotuerk if the `ASSERT_EQ` above fails, the typesupport struct will be left in a broken state.

Reply:
See https://github.com/ros2/rmw_implementation/pull/152/commits/af19cb4dce63ec866d9597589c694277feba8a3f.